### PR TITLE
Replace input buffer with scratch buffer

### PR DESCRIPTION
**Pull request by ascjones**
Tested with examples/lang/erc20 - works.

Rel: https://github.com/paritytech/substrate/pull/2911

### CODE REVIEW

File: core/src/env/srml/srml_only/impls.rs [hunk 1]
```diff
@@ -113,17 +113,7 @@ where
...
-            Vec::new()
-        } else {
-            let mut buffer = Vec::new();
-            buffer.resize(size as usize, 0);
-            unsafe {
-                sys::ext_input_copy(buffer.as_mut_ptr() as u32, 0, size);
-            }
-            buffer
-        }
+        read_scratch_buffer()
```
Comment (Robbepop) [inline core/src/env/srml/srml_only/impls.rs]: I like the simplicity of this change. :+1:

**Comment by xlc** [comment]
does this means ink master no long works with substrate v1.0?

**Comment by ascjones** [comment]
@xlc correct

**Comment by xlc** [comment]
any plan to maintain compatibility with substrate v1.0? substrate master are discouraged to use for development due to frequent breaking changes
Related #95

**Comment by ascjones** [comment]
It's a good point. Just spoke to @Robbepop about this and we will look into implementing versioning when he returns.

**Pull request merged**